util/rlp/src/impls.rs

```diff
@@ -0,0 +1,120 @@
+use byteorder::{WriteBytesExt, BigEndian};
+use bigint::prelude::{Uint, U128, U256, H64, H128, H160, H256, H512, H520, H2048};
+use traits::Encodable;
+use stream::RlpStream;
+
+impl Encodable for bool {
+	fn rlp_append(&self, s: &mut RlpStream) {
+		if *self {
+			s.encoder().encode_value(&[1]);
+		} else {
+			s.encoder().encode_value(&[0]);
+		}
+	}
+}
+
+impl<'a> Encodable for &'a [u8] {
+	fn rlp_append(&self, s: &mut RlpStream) {
+		s.encoder().encode_value(self);
+	}
+}
+
+impl Encodable for Vec<u8> {
+	fn rlp_append(&self, s: &mut RlpStream) {
+		s.encoder().encode_value(self);
+	}
+}
+
+impl<T> Encodable for Option<T> where T: Encodable {
```

Comment:
Manually. My point is that changing e.g. a struct member type `T` to `Option<T>` silently changes encoding of that member to a list. I recall fixing a bug caused by this.